contrib/buildsystems/CMakeLists.txt

```diff
@@ -58,6 +58,10 @@ if(WIN32)
 
 	# In the vcpkg edition, we need this to be able to link to libcurl
 	set(CURL_NO_CURL_CMAKE ON)
+
+	# Copy the necessary vcpkg DLLs (like iconv) to the install dir
+	set(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)
+	set(CMAKE_TOOLCHAIN_FILE ${VCPKG_DIR}/scripts/buildsystems/vcpkg.cmake CACHE STRING "Vcpkg toolchain file")
```

Comment:
This causes the [build](https://github.com/git-for-windows/git/pull/2971/checks?check_run_id=1671547046#step:7:13) to fail:

```
 CMake Error at C:/Program Files/CMake/share/cmake-3.19/Modules/CMakeDetermineSystem.cmake:110 (message):
  Could not find toolchain file:
  D:/a/git/git/contrib/buildsystems/../../compat/vcbuild/vcpkg/scripts/buildsystems/vcpkg.cmake
```

Reply:
It works locally, so something is missing in your vcpkg artifacts in GH Actions. I just checked:

Downloaded https://dev.azure.com/git/4c888a9b-b30a-4c62-9301-8dfa18a71976/_apis/build/builds/2101/artifacts?artifactName=compat&api-version=6.0&%24format=zip

![image](https://user-images.githubusercontent.com/17739158/104073593-64da9b00-520e-11eb-9827-6e5e0884f988.png)

Could you please include `vcpkg/scripts/buildsystems/vcpkg.cmake` in your vckpg artifacts, @dscho? That should resolve the issue 👍🏼

Reply:
Hrm. I did that, and now the `vs-build` jobs take longer than even the `linux-gcc` build (which builds Git on Linux and then runs the test suite _twice_). To test the hypothesis that this is triggered by including `scripts`, I started https://github.com/dscho/git/actions/runs/474627271. Let's see how that fares. If its `vs-build` jobs also take excruciatingly long, I will have to roll back that `scripts` change.

Reply:
Nope, seems that https://github.com/dscho/git/runs/1674724825?check_suite_focus=true was done with the `generate Visual Studio solution` step within 1.5 minutes, but https://github.com/git-for-windows/git/pull/2971/checks?check_run_id=1674658393 has now been running for more than 30 minutes.

Reply:
That's weird. Looks like it just wasn't doing anything for 6 (!) hours. It's probably better if we first clone `vcpkg` in CI and then download your artifacts over it. I'll try that tomorrow evening if I can make it work time-wise. To be continued 👍🏼

Reply:
It really is a bit strange. All I wanted to do in the `vcpkg` build was to have pre-generated artifacts, and it _looked_ as if they were being picked up correctly via CMake even without the `scripts/` folder.

So I just kicked off another build that excludes the `scripts/` folder again. We may have to resort to uglier methods such as hard-coding some `install()` steps based on constants defined via the `find_package()` statements.

Reply:
From my experience CMAKE_TOOLCHAIN_FILE should be defined from the command line as it's run before the actual CMAKE file.  it's general purpose is to configure CMake for cross-compilation.  [From the official docs](https://cmake.org/cmake/help/latest/variable/CMAKE_TOOLCHAIN_FILE.html)

> **This variable is specified on the command line when cross-compiling with CMake.** It is the path to a file which is read early in the CMake run and which specifies locations for compilers and toolchain utilities, and other target platform and compiler related information.

emphasis mine. 

If this toolchain file is required for any build I recommend switching this out for checking the variables that this would state.

Reply:
I don't even _know_ what purpose this toolchain file serves.

Reply:
I tried without the `CMAKE_TOOLCHAIN_FILE` but then it doesn't copy over the DLLs. So it is needed. But I was able to fix CI in https://github.com/git-for-windows/git/pull/2971/commits/d8cf02ee688cbf781bf74f9d6f33ab2fea8493fc, how does that look @dscho? 😊

Reply:
Doesn't that take super long because it rebuilds `vcpkg.exe` every single time? I'd rather have a manual copy step, using constants defined by `find_package()`, if that is at all possible.

Reply:
1m51, that is: https://github.com/git-for-windows/git/pull/2971/checks?check_run_id=1687684942. But I see that `vcpkg.exe` is also included in your artifact, so let me try to exclude the `vcpkg.exe` build step and see if that works

Reply:
Fixed in https://github.com/git-for-windows/git/pull/2971/commits/09d15737479ea71d9656e5383d41a450f474ce35. Now the new step `initialize vcpkg` only takes 6s 👍🏼